java/Cherepnin_Oleh/src/com/zoolatech/lecture3/tasks/_2/NameValidator.java

```diff
@@ -0,0 +1,30 @@
+package com.zoolatech.lecture3.tasks._2;
+
+import java.lang.reflect.Field;
+
+public class NameValidator implements SmallerValidator {
+
+    private String regex = "[A-Z][a-z]+";
+
+    @Override
+    public boolean isValid(User user) {
+        try {
+            boolean firstNameCheck = printMessage(user.getClass().getDeclaredField("firstName"), user.getFirstName());
+            boolean lastNameCheck = printMessage(user.getClass().getDeclaredField("lastName"), user.getLastName());
+            boolean countryCheck = printMessage(user.getClass().getDeclaredField("country"), user.getCountry());
+            return firstNameCheck && lastNameCheck && countryCheck;
+        } catch (NoSuchFieldException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    private boolean printMessage(Field field, String name) {
+        if (name.isBlank()) {
+            System.out.println(field.getName() + " is blank");
+        } else if (!name.matches(regex)) {
```

Comment:
It's better to create an instance of the class Pattern and reuse it instead of using `String.matches`. Method `matches` will create a new instance of the Pattern during every invocation, and this is a heavy operation.